tasks/release.js

```diff
@@ -49,88 +46,113 @@ function pack () {
 }
 
 /**
- * Use electron-packager to build electron app
+ * Use electron-builder to build electron app
  */
-function build () {
-  let options = require('../config').building
+function build (platform = process.platform, arch = process.arch) {
+  let config = require('../config').building
 
-  options.afterCopy = [
+  // defined build target
+  let electronPlatform = platform
+  if (platform === 'win32') {
+    electronPlatform = 'win'
+  }
+  let targets = {
+    [arch]: true,
+    [electronPlatform]: []
+  }
+
+  // icons need to be available in the dist folder
+  copyIcons()
+
+  config.afterPack = ({outDir, appOutDir}) => {
     binaryPath
-      ? copyBinary('gaia', binaryPath)
-      : buildGaiaBinary()
-  ]
-  // prune installs the packages
-  options.afterPrune = [
-    // we need to rebuild some native packages for the electron environment
-    function rebuildNodeModules (buildPath, electronVersion, platform, arch, callback) {
-      rebuild({ buildPath, electronVersion, arch })
-        .then(callback)
-        .catch(callback)
-    }
-  ]
+    ? copyBinary('gaia', binaryPath, platform)({outDir, appOutDir})
+    : goBuild(`github.com/cosmos/gaia/cmd/gaia`, platform, arch)({outDir, appOutDir})
+    copyNetworks(appOutDir)
+  }
 
   console.log('\x1b[34mBuilding electron app(s)...\n\x1b[0m')
-  packager(options, (err, appPaths) => {
-    if (err) {
-      console.error('\x1b[31mError from `electron-packager` when building app...\x1b[0m')
-      console.error(err)
-    } else {
+  // Promise is returned
+  return builder.build({
+    ...targets,
+    config
+  })
+    .then(() => {
       console.log('Build(s) successful!')
-      console.log(appPaths)
-
       console.log('\n\x1b[34mDONE\n\x1b[0m')
-    }
-  })
-}
-
-function copyBinary (name, binaryLocation) {
-  return function (buildPath, electronVersion, platform, arch, cb) {
-    let binPath = join(buildPath, 'bin', name)
-    if (platform === 'win32') {
-      binPath = binPath + '.exe'
-    }
-    fs.copySync(binaryLocation, binPath)
-    cb()
-  }
+    })
+    .catch((error) => {
+      console.error('\x1b[31mError from `electron-packager` when building app...\x1b[0m')
+      console.error(error)
+    })
 }
 
+/*
+* Build the baseserver binary. Should only be done in development!
+*/
 const GOARCH = {
   'x64': 'amd64',
   'ia32': '386'
 }
-
-function buildGaiaBinary () {
-  return function (buildPath, electronVersion, platform, arch, cb) {
+function goBuild (pkg, platform, arch) {
+  return function ({outDir, appOutDir}) {
     if (platform === 'win32') platform = 'windows'
     if (platform === 'mas') platform = 'darwin'
     if (GOARCH[arch]) arch = GOARCH[arch]
 
     console.log(`\x1b[34mBuilding gaia binary (${platform}/${arch})...\n\x1b[0m`)
 
-    let output = 'gaia'
-    if (platform === 'windows') output += '.exe'
-
-    let cmd = `
```

Comment:
Looks like the deterministic gaia build from #314 got removed (I could make a PR on top of this PR to re-enable it).

Reply:
Some merge conflicts probably. Sorry. Readded your 2(?) commits. But it fails on Windows:
![image](https://user-images.githubusercontent.com/5869273/34692707-58c0c932-f4c1-11e7-912d-22b93f92a35d.png)

Reply:
I removed the commits again. Could create a PR like you proposed? My thoughts: Do you think this is actually necessary? In development we don't need deterministic builds right? And in production I propose we always use a distributed binary with the "--binary" flag.

Reply:
>But it fails on Windows:

Ah OK I'll fix it, just need the Windows equivalent to `/tmp`.

>My thoughts: Do you think this is actually necessary? In development we don't need deterministic builds right? And in production I propose we always use a distributed binary with the "--binary" flag.

It's the difference between manually downloading the gaia binary every time we build, or instead having our build system make the binary for us. This way, we will all be independently building gaia from source, instead of maintaining a single point of failure (plus we don't have to wait for someone else to make binaries for us).

Reply:
i see both sides. not sure i can vote one way or another. sorry guys.

Reply:
I vote for using a distributed Gaia binary for simplicity. We can add multiplatform Gaia binary building in another PR, but I would love to get this `electron-builder` PR merged.